### PR TITLE
Fix #318, updated the whole process in order to use our repository

### LINKED ISSUE #318: Update the process in order to use the packages stored in our new repository instead of the online ones

### PR DESCRIPTION
**Pull request by giuseppe-carboni**
All dependencies original URLs have been removed from this repository.
They are now stored in a `sources` text file inside the
discos/dependencies repository (which is private). The process now asks
the user to download part of the dependencies repository with their
github credentials. A token for that repository gets downloaded and
stored the first time the provisioning process is launched. That token
is then used to download single packages from the dependencies repository.
Also, minor bugs were fixed.

### CODE REVIEW

File: ansible/group_vars/all [hunk 1]
```diff
@@ -10,3 +10,4 @@ observer: { name: "observer", uid: 3061, group: "{{ users_groups.acs }}", admini
 discos_sw_dir: "discos-sw"
 
 local_repository_path: "{{ lookup('env','HOME') }}/.deployment/repository"
+remote_repository_download_url: "https://raw.github.com/discos/dependencies/dependencies"
```
Comment (marco-buttu) [inline ansible/group_vars/all]: To change: `dependencies` to `discos-provisioning`. So we can create more branches, one for the dependencies required by every repository

File: ansible/roles/acs/tasks/discos_dependencies.yml [hunk 2]
```diff
@@ -46,17 +46,18 @@
 
 - name: Check if the fitsviewer binary exist in the local repository
   stat:
-    path: "{{ local_repository_path }}/{{ fv_file }}"
+    path: "{{ local_repository_path }}/{{ fv.file }}"
   delegate_to: localhost
   run_once: true
```
Comment (marco-buttu) [inline ansible/roles/acs/tasks/discos_dependencies.yml]: True?

File: ansible/roles/acs/tasks/discos_dependencies.yml [hunk 3]
```diff
@@ -66,21 +67,23 @@
...
-    url: "http://heasarc.gsfc.nasa.gov/fitsio/CCfits-2.4/{{ ccfits.file }}"
+    url: "{{ remote_repository_download_url }}/{{ ccfits.file }}"
     dest: "{{ local_repository_path }}"
+    headers:
+      Authorization: "token {{ repository_token }}"
   when:
     - ccfits_include.stat.exists == False
     - ccfits_sources.stat.exists == False
   delegate_to: localhost
   run_once: true
```
Comment (marco-buttu) [inline ansible/roles/acs/tasks/discos_dependencies.yml]: True?

File: ansible/roles/acs/tasks/discos_dependencies.yml [hunk 4]
```diff
@@ -351,19 +356,21 @@
     path: "/usr/local/include/f2c.h"
   register: f2c_header
 
+
 - name: Check if the f2c sources exist in the local repository
   stat:
-    path: "{{ local_repository_path }}/f2c.zip"
+    path: "{{ local_repository_path }}/{{ f2c.file }}"
   delegate_to: localhost
   run_once: true
```
Comment (marco-buttu) [inline ansible/roles/acs/tasks/discos_dependencies.yml]: True?

File: ansible/roles/acs/tasks/discos_dependencies.yml (hunk 2 above)
Comment (giuseppe-carboni) [reply ansible/roles/acs/tasks/discos_dependencies.yml]: Both true and True are fine for Ansible, you can also use Yes or yes, sometimes I change the old `true` variables to `True` in order to be more consistent when I update parts of the code, but seems like this time I didn't update them. If we want to keep this kind of consistency too we should open a issue just for this.

**Pull request merged**